### PR TITLE
Refine segment stat metrics on QueryNode

### PR DESCRIPTION
**Pull request by soothing-rain**
issue: #20760

/kind improvement

Signed-off-by: Yuchen Gao <yuchen.gao@zilliz.com>

### CODE REVIEW

File: internal/util/typeutil/map.go [hunk 1]
```diff
@@ -35,26 +39,25 @@ func GetMapKeys(src map[string]string) []string {
...
 }
 
-func (m *ConcurrentMap[K, V]) Get(key K) (V, bool) {
-	var zeroValue V
-	value, ok := m.inner.Load(key)
-	if !ok {
-		return zeroValue, ok
+func (m *ConcurrentMap[K, V]) Insert(key K, value V) {
+	if _, exist := m.inner.LoadOrStore(key, value); exist {
+		m.len.Inc()
```
Comment (yah01) [inline internal/util/typeutil/map.go:56]: If the key exists, no new element inserted, no change to the number of elements

File: internal/util/typeutil/map.go [hunk 2]
```diff
@@ -67,20 +70,18 @@ func (m *ConcurrentMap[K, V]) GetOrInsert(key K, value V) (V, bool) {
 	var zeroValue V
 	loaded, exist := m.inner.LoadOrStore(key, value)
 	if !exist {
+		m.len.Inc()
 		return zeroValue, exist
```
Comment (yah01) [inline internal/util/typeutil/map.go:74]: should return value, exist here

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**